<?xml version="1.0"?>
<launch>
  <!-- Arguments -->

  <arg name="publish_gesture_topic" default="/gesture/hand_sign"/>
  <arg name="control_topic" default="/robot_diff_drive_controller/cmd_vel"/>
  <arg name="camera_name" default="/dev/video0"/>
  <arg name="drive_robot" default="false"/>

  <!-- Camera Node -->
  <node name="my_webcam" pkg="hand_steer_sim" type="image_publisher_launch.py" output="screen">
    <param name="camera_name" type="string" value="$(arg camera_name)" />
  </node>  

  <!-- Optional Drive Robot -->
  <group if="$(arg drive_robot)">
    <include file="$(find hand_steer_sim)/launch/drive_robot.launch" />
  </group>
  
  <!-- Hand Gesture Recognition -->
  <include file="$(find hand_steer_sim)/launch/gesture_recognition.launch" />
    
  <!-- Gesture to Controller Conversion -->
  <node name="sign_to_controller" pkg="hand_steer_sim" type="sign_to_controller.py" output="screen">
    <param name="publish_gesture_topic" type="string" value="$(arg publish_gesture_topic)" />
    <param name="control_topic" type="string" value="$(arg control_topic)" />
  </node>
</launch>
